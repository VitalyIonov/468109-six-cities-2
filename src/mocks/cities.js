export const Cities = [
  {
    name: `Dusseldorf`,
    location: {latitude: 51.225402, longitude: 6.776314, zoom: 13}
  },
  {
    name: `Paris`,
    location: {latitude: 51.225402, longitude: 6.776314, zoom: 13}
  },
  {
    name: `Cologne`,
    location: {latitude: 51.225402, longitude: 6.776314, zoom: 13}
  },
  {
    name: `Amsterdam`,
    location: {latitude: 51.225402, longitude: 6.776314, zoom: 13}
  },
];
