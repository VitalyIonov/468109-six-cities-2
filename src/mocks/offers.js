export const Offers = [
  {
    id: 1,
    image: {
      src: `img/room.jpg`,
      width: `260`,
      height: `200`,
      alt: `Place image`
    },
    price: `80`,
    period: `night`,
    rating: `80%`,
    coordinates: [52.3909553943508, 4.85309666406198],
    description: `Beautiful & luxurious apartment at great location`,
    type: `Private room`
  },
  {
    id: 2,
    image: {
      src: `img/room.jpg`,
      width: `260`,
      height: `200`,
      alt: `Place image`
    },
    price: `180`,
    period: `night`,
    rating: `60%`,
    coordinates: [52.369553943508, 4.85309666406198],
    description: `Wood and stone place`,
    type: `Private room`
  },
  {
    id: 3,
    image: {
      src: `img/room.jpg`,
      width: `260`,
      height: `200`,
      alt: `Place image`
    },
    price: `820`,
    period: `night`,
    rating: `30%`,
    coordinates: [52.3909553943508, 4.929309666406198],
    description: `Canal View Prinsengracht`,
    type: `Private room`
  },
  {
    id: 4,
    image: {
      src: `img/room.jpg`,
      width: `260`,
      height: `200`,
      alt: `Place image`
    },
    price: `81`,
    period: `night`,
    rating: `100%`,
    coordinates: [52.3809553943508, 4.939309666406198],
    description: `Nice, cozy, warm big bed apartment`,
    type: `Private room`
  }
];
