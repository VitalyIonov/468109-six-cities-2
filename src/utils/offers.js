export const getOffersByCity = (city, offers) => offers.filter((offer) => offer.city.name === city.name);

export const getCitiesList = (offers) => {
  const MAX_CITIES = 6;
  const stringifiedCities = [...new Set(offers.map((offer) => JSON.stringify(offer.city)))];
  const cities = stringifiedCities.map((city) => JSON.parse(city));

  if (cities.length > MAX_CITIES) {
    cities.length = MAX_CITIES;
  }

  return cities;
};

export const formatToClient = (offers) => offers.map((offer) => Object.assign({}, offer, {
  previewImage: offer.preview_image || offer.previewImage
}));
