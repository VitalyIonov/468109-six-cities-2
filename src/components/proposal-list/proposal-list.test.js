import React from 'react';
import renderer from 'react-test-renderer';
import ProposalList from './proposal-list';

import {Offers} from '../../mocks/offers';

it(`ProposalList correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<ProposalList
      data={Offers}
      onPlaceTitleClick={jest.fn()}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
