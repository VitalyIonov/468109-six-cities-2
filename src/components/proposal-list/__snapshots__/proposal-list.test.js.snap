// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposalList correctly renders after relaunch 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Canal View Prinsengracht"
          className="place-card__image"
          src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            571
          </b>
          <span
            className="place-card__price-text"
          >
            / 
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": 44,
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[MockFunction]}
      >
        <a
          href="#"
        >
          Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Canal View Prinsengracht"
          className="place-card__image"
          src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            571
          </b>
          <span
            className="place-card__price-text"
          >
            / 
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": 56,
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[MockFunction]}
      >
        <a
          href="#"
        >
          Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Canal View Prinsengracht"
          className="place-card__image"
          src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/14.jpg"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            571
          </b>
          <span
            className="place-card__price-text"
          >
            / 
          </span>
        </div>
        <button
          className="place-card__bookmark-button place-card__bookmark-button--active button"
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            In bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": 96,
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[MockFunction]}
      >
        <a
          href="#"
        >
          Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
</div>
`;
